test/lib/RuleResultRecombinator.test.ts

```diff
@@ -0,0 +1,199 @@
+	{
+		engine: 'eslint',
+		fileName: sampleFile1,
+		violations: [{
+			"line": 2,
+			"column": 11,
+			"severity": 2,
+			"message": "'unusedParam' is defined but never used.",
+			"ruleName": "no-unused-vars",
+			"category": "Variables",
+			"url": "https://eslint.org/docs/rules/no-unused-vars"
+		}]
+	},
+	{
+		engine: 'eslint',
+		fileName: sampleFile2,
+		violations: [{
+			"line": 4,
+			"column": 11,
+			"severity": 2,
+			"message": "'unusedParam' is defined but never used.",
+			"ruleName": "no-unused-vars",
+			"category": "Variables",
+			"url": "https://eslint.org/docs/rules/no-unused-vars"
+		}, {
+			"line": 6,
+			"column": 9,
+			"severity": 2,
+			"message": "'unusedVar' is assigned a value but never used.",
+			"ruleName": "no-unused-vars",
+			"category": "Variables",
+			"url": "https://eslint.org/docs/rules/no-unused-vars"
+		}]
+	},
+	{
+		engine: "pmd",
+		fileName: sampleFile3,
+		violations: [{
+			"line": 2,
+			"column": 1,
+			"endLine": 2,
+			"endColumn": 57,
+			"severity": 4,
+			"ruleName": "UnusedImports",
+			"category": "Best Practices",
+			"url": "https://pmd.github.io/pmd-6.22.0/pmd_rules_java_bestpractices.html#unusedimports",
+			"message": "\nAvoid unused imports such as 'sfdc.sfdx.scanner.messaging.SfdxMessager'\n"
+		}, {
+			"line": 4,
+			"column": 8,
+			"endLine": 56,
+			"endColumn": 1,
+			"severity": 3,
+			"ruleName": "CommentRequired",
+			"category": "Documentation",
+			"url": "https://pmd.github.io/pmd-6.22.0/pmd_rules_java_documentation.html#commentrequired",
+			"message": "\nEnum comments are required\n"
+		}, {
+			"line": 5,
+			"column": 1,
+			"endLine": 5,
+			"endColumn": 2,
+			"severity": 3,
+			"ruleName": "CommentSize",
+			"category": "Documentation",
+			"url": "https://pmd.github.io/pmd-6.22.0/pmd_rules_java_documentation.html#commentsize",
+			"message": "\nComment is too large: Line too long\n"
+		}]
+	}
+];
+
+function isString(x: string | {columns; rows}): x is string {
+	return typeof x === 'string';
+}
+
+
+describe('RuleResultRecombinator', () => {
+	describe('#recombineAndReformatResults()', () => {
+		describe('Output Format: JUnit', () => {
+			// This is a function for validating JUnit-formatted XMLs.
+			function validateJUnitFormatting(lines: string[], fileNames: string[], violationCounts: number[]) {
+				expect(fileNames.length).to.equal(violationCounts.length,
+					`Improperly constructed rule. Supplied ${fileNames.length} names and ${violationCounts.length} violation counts`
+				);
+				// We expect the first line to be the <testsuites> opening tag.
+				expect(lines[0]).to.equal('<testsuites>', 'First line should have been the testsuites opening tag');
+				// After that, we expect to see a pattern of <testsuite> tags wrapping at least one <testcase> tag,
+				// each of which wraps exactly one <failure> tag.
+				let lineCounter = 1;
+				let idx = 0;
+				while (idx < fileNames.length) {
+					let fileName = fileNames[idx];
```

Comment:
This file seems to use `let` where it could be `const`.

Reply:
Good point. I'll fix that.